relay-server/src/actors/store.rs

```diff
@@ -176,6 +135,60 @@ fn make_producer<'a>(
 }
 
 impl StoreForwarder {
+    pub fn start(self) -> Addr<Self> {
+        relay_log::info!("store forwarder started");
+
+        let (tx, mut rx) = mpsc::unbounded_channel::<StoreMessageEnvelope>();
+
+        tokio::spawn(async move {
+            while let Some(message) = rx.recv().await {
+                match message {
+                    StoreMessageEnvelope::StoreEnvelope(msg, responder_tx) => {
+                        let response = self.handle_store_evelope(msg);
+                        responder_tx.send(response).ok();
+                    }
+                }
+            }
+
+            relay_log::info!("store forwarder stopped");
+        });
+
+        Addr { tx }
+    }
+
+    pub fn create(config: Arc<Config>) -> Result<Self, ServerError> {
+        let mut reused_producers = BTreeMap::new();
+        let producers = Producers {
```

Comment:
As we're moving code already, could you move this into a `Producers::create()` method?

Reply:
The entire function or just the code after line 161 (and then do `let producers = Producers::create()`)?